src/Workspaces/CSharp/Portable/Classification/SyntaxClassification/DiscardSyntaxClassifier.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Collections.Immutable;
+using System.Threading;
+using Microsoft.CodeAnalysis.Classification;
+using Microsoft.CodeAnalysis.Classification.Classifiers;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Classification.Classifiers
+{
+    internal class DiscardSyntaxClassifier : AbstractSyntaxClassifier
+    {
+        public override ImmutableArray<Type> SyntaxNodeTypes { get; } = ImmutableArray.Create(
+            typeof(DiscardDesignationSyntax),
+            typeof(DiscardPatternSyntax),
+            typeof(ParameterSyntax),
+            typeof(IdentifierNameSyntax));
+
+        public override void AddClassifications(
+           Workspace workspace,
+           SyntaxNode syntax,
+           SemanticModel semanticModel,
+           ArrayBuilder<ClassifiedSpan> result,
+           CancellationToken cancellationToken)
+        {
+            if (syntax.IsKind(SyntaxKind.DiscardDesignation) || syntax.IsKind(SyntaxKind.DiscardPattern))
+            {
+                result.Add(new ClassifiedSpan(syntax.Span, ClassificationTypeNames.Keyword));
+                return;
+            }
+
+            switch (syntax)
+            {
+                case ParameterSyntax parameter:
+                    var symbol = semanticModel.GetDeclaredSymbol(parameter, cancellationToken);
+
+                    if (symbol.IsDiscard)
```

Comment:
`DiscardSyntaxClassifier.cs(42,25): error CS8602: Dereference of a possibly null reference.`

Reply:
@JoeRobich i thought about that at first but was then confused by this: 
![image](https://user-images.githubusercontent.com/13168856/72022280-53b02480-3270-11ea-8a30-d3c9e95db5d7.png)

Reply:
anyways i added the requested change in https://github.com/dotnet/roslyn/pull/40396/commits/7bdbfb987166d5741350f74b443b91e1f885d7f6

Reply:
Looks like nullable annotations (https://github.com/dotnet/roslyn/commit/f19813a8b19d3148c4602fec227eaaf686492d3b#diff-a5b06c0c9cabe56b6ceae67f7aa0f3c7R1431) were committed to the master branch since you started this work. The CI runs with those changes merged with your PR.

Reply:
Ahh ok. thanks :)